examples/ssr-islands-axum/src/app.rs

```diff
@@ -0,0 +1,139 @@
+
+static_loader! {
+    static ISLANDS_TRANSLATIONS = {
+        locales: "./locales/islands",
+        fallback_language: "en",
+    };
+}
+
+#[macro_export]
+macro_rules! i18n {
+    ($translations:expr, $locales:expr$(,)?) => {{
+        ::leptos_fluent::leptos_fluent! {
+            translations: $translations,
+            languages: "./locales/languages.json",
+            locales: $locales,
+            sync_html_tag_lang: true,
+            sync_html_tag_dir: true,
+            cookie_name: "lang",
+            cookie_attrs: "SameSite=Strict; Secure; path=/; max-age=600",
+            initial_language_from_cookie: true,
+            initial_language_from_cookie_to_localstorage: true,
+            set_language_to_cookie: true,
+            url_param: "lang",
+            initial_language_from_url_param: true,
+            initial_language_from_url_param_to_localstorage: true,
+            initial_language_from_url_param_to_cookie: true,
+            set_language_to_url_param: true,
+            localstorage_key: "language",
+            initial_language_from_localstorage: true,
+            initial_language_from_localstorage_to_cookie: true,
+            set_language_to_localstorage: true,
+            initial_language_from_navigator: true,
+            initial_language_from_navigator_to_localstorage: true,
+            initial_language_from_accept_language_header: true,
+        }
+    }};
+}
+
+#[component]
+pub fn App() -> impl IntoView {
+    // Provides context that manages stylesheets, titles, meta tags, etc.
+    provide_meta_context();
+
+    i18n!([COMPONENTS_TRANSLATIONS], "./locales/server");
+
+    view! {
+        <Stylesheet id="leptos" href="/pkg/leptos-fluent-ssr-hydrate-axum-islands-example.css" />
+
+        <Title text=tr!("welcome-to-leptos") />
+
+        <Router fallback=|| {
+            let mut outside_errors = Errors::default();
+            outside_errors.insert_with_default_key(AppError::NotFound);
+            view! { <ErrorTemplate outside_errors /> }.into_view()
+        }>
+            <main>
+                <Routes>
+                    <Route path="" view=HomePage />
+                </Routes>
+            </main>
+        </Router>
+    }
+}
+
+#[component]
+pub fn HomePage() -> impl IntoView {
+    view! {
+        <Archipelago>
+            <Counter />
+            <LanguageSelector />
+        </Archipelago>
+    }
+}
+
+#[island]
+fn Archipelago(children: Children) -> impl IntoView {
+    i18n!([ISLANDS_TRANSLATIONS], "./locales/islands");
+
+    // Children will be executed when the i18n context is ready.
+    // See https://book.leptos.dev/islands.html#passing-context-between-islands
+    //
+    // > That’s why in `HomePage`, I made `let tabs = move ||` a function, and
+    // > called it like `{tabs()}`: creating the tabs lazily this way meant that
+    // > the `Tabs` island would already have provided the `selected` context by
+    // > the time each `Tab` went looking for it.
+    children()
```

Comment:
> Is there a problem with the way I created the parent island and added the children islands?

This is the answer. See the final note in [*Passing Context Between Islands*](https://book.leptos.dev/islands.html#passing-context-between-islands):

> That’s why in `HomePage`, I made `let tabs = move ||` a function, and called it like `{tabs()}`: creating the tabs lazily this way meant that the `Tabs` island would already have provided the `selected` context by the time each `Tab` went looking for it.

Is really tricky and very contraintuitive.